src/mpi/romio/adio/ad_daos/ad_daos_close.c

```diff
@@ -28,8 +28,10 @@ void ADIOI_DAOS_Close(ADIO_File fd, int *error_code)
     adio_daos_poh_release(cont->p);
     cont->p = NULL;
 
-    ADIOI_Free(cont->obj_name);
-    ADIOI_Free(cont->cont_name);
+    if (rank == 0) {
```

Comment:
Where can I find the corresponding change that shows this allocation only happening on rank 0? I'm not able to find it in the commit at the moment.

Reply:
Those allocations happen in ADIOI_DAOS_Open()
ADIOI_DAOS_Open() is called only on rank 0 from ADIOI_DAOS_OpenColl():

https://github.com/pmodels/mpich/pull/4816/files?file-filters%5B%5D=.c#diff-625412bb51f58d78218049fa75b857ccR459

Reply:
The corresponding change is here:
https://github.com/pmodels/mpich/pull/4816/files#diff-625412bb51f58d78218049fa75b857ccL749

Reply:
OK, got it.